spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java

```diff
@@ -374,32 +383,31 @@ void terminateSpanOnCancel() {
 							+ this.span + "]");
 				}
 
-				this.span.tag("error", CANCELLED_SUBSCRIPTION_ERROR);
-				handleReceive(this.span, null, null);
+				handleReceive(this.span, null, CANCELLED_ERROR);
 			}
 
 			void terminateSpan(@Nullable ClientResponse clientResponse,
-					@Nullable Throwable throwable) {
+					@Nullable Throwable error) {
 				if (clientResponse == null) {
 					if (log.isDebugEnabled()) {
 						log.debug("No response was returned. Will close the span ["
 								+ this.span + "]");
 					}
-					handleReceive(this.span, clientResponse, throwable);
+					handleReceive(this.span, null, error);
 					return;
 				}
 				int statusCode = clientResponse.rawStatusCode();
-				boolean error = statusCode >= 400;
-				if (error) {
+				boolean isHttpError = statusCode >= 400;
```

Comment:
ps this isn't correct either, but I'm not changing this now. later we should redo most of this instrumentation